workshops/views.py

```diff
@@ -178,6 +178,16 @@ class SiteUpdate(LoginRequiredMixin, UpdateViewContext):
     slug_url_kwarg = 'site_domain'
     template_name = 'workshops/generic_form.html'
 
+
+@login_required
+def site_delete(request, site_domain):
+    """Delete specific site."""
+    site = get_object_or_404(Site, domain=site_domain)
+    site.delete()
+
+    messages.success(request, 'Site was deleted successfully.')
```

Comment:
I don't understand your concerns about messages framework. It's a pretty standard way of displaying results of backend operations to the enduser. It came with #352.

> sounds like a lot of state to store in the expectation that you're going to get a particular future request

With `return redirect()` we will get it.

messages are invalidated as soon as they're traversed through, which happens in every view that uses our original template.